### ISSUE TITLE
`sort date` uses month and year fields instead

### ISSUE THREAD

**Issue by gaomy3832** [pcooksey/bibtex-js#48]
Currently if I use
```
<div class="sort date" extra="DESC date">
  <div class="templates"></div>
</div>
```
The bibtex entries have to have a `date` field to provide the date. When the entry does not have the `date` field, is there a way to use the `month` and `year` fields to derive the date as the sort key?

I quickly scan through the code, and think this feature could be easily added into the `sortArray` function at https://github.com/pcooksey/bibtex-js/blob/5ccf967e3afb0e319b14b12a6e3a442be932bf0a/src/bibtex_js.js#L627

I can think of two ways: 1. add a new sort type called something like `monthyear`, and get the value from those fields. 2. reuse the current sort type `date`, and compose the value from month and year if date is not presented.

If you have a preference and let me know, I can compose a quick pull request to add this feature.

**Comment by pcooksey** [comment]
Well for the 1st option it looks like `monthyear` would imply that the `date` field isn't used at all. So then we'd have the same problem where some entries only have `date` and not the combined `monthyear`.

I think the 2nd option is probably the best. I think if the user wants to sort by keyword `date` then falling back on the `month` and `year` would be preferable. Of course, if the user is sorting by keyword `year` then they might also want to fall back on the year in the `date` field. 

If you'd like to work on it that would be great! I will need some test-cases before committing it to the repo to ensure everything is working and to ensure functionality when making changes in the future.

**Comment by gaomy3832** [comment]
This issue has been resolved in #50.

**Comment by pcooksey** [comment]
Closing with commits 1705bfca24e0b1fddbc02b4f6cdcbd00e22af1ac and 3079ab336b1cc75f1229bdf8f1292b11d644e7ec
